AutoRest/Generators/CSharp/CSharp/Templates/MethodTemplate.cshtml

```diff
@@ -52,9 +62,9 @@ public async Task<@(Model.OperationResponseReturnTypeString)> @(Model.Name)WithH
     {
         invocationId = ServiceClientTracing.NextInvocationId.ToString();
         Dictionary<string, object> tracingParameters = new Dictionary<string, object>();
-    @foreach (var parameter in Model.LocalParameters)
+    @foreach (var parameter in Model.ParameterTemplateModels.Where(p => !p.IsParameterGroup && p.ClientProperty == null))
```

Comment:
@yugangw-msft I updated the getter to return all parameters *without* a parameter group.  That set would *include* the ParameterGroup parameter itself.
```C#
ParameterTemplateModels.Where(
                        p => p != null && p.ClientProperty == null && string.IsNullOrEmpty(p.ParameterGroup) && !string.IsNullOrWhiteSpace(p.Name))
                        .OrderBy(item => !item.IsRequired);
```

Here, I am getting all the parameters (even if they were grouped) and excluding just the "fake" parameters (the ParameterGroup parameter itself)
```C#
Model.ParameterTemplateModels.Where(p => !p.IsParameterGroup && p.ClientProperty == null)
```

So it's really two different cases -- this change isn't redundant.